Extensions/Spine/spineruntimeobject.ts

```diff
@@ -0,0 +1,222 @@
+namespace gdjs {
+  type SpineAnimation = { name: string; source: string; loop: boolean };
+
+  export type SpineObjectDataType = {
+    content: {
+      opacity: float;
+      scale: float;
+      timeScale: float;
+      jsonResourceName: string;
+      atlasResourceName: string;
+      imageResourceName: string;
+      animations: SpineAnimation[];
+    };
+  };
+  export type SpineObjectData = ObjectData & SpineObjectDataType;
+
+  export class SpineRuntimeObject extends gdjs.RuntimeObject {
+    private _opacity: float;
+    private _scale: number;
+    private _timeScale: number;
+    private _animations: SpineAnimation[];
+    private _currentAnimationIndex = 0;
+    private _renderer: gdjs.SpineRuntimeObjectPixiRenderer;
+
+    readonly jsonResourceName: string;
+    readonly atlasResourceName: string;
+    readonly imageResourceName: string;
+
+    /**
+     * @param instanceContainer The container the object belongs to.
+     * @param objectData The object data used to initialize the object
+     */
+    constructor(
+      instanceContainer: gdjs.RuntimeInstanceContainer,
+      objectData: SpineObjectData
+    ) {
+      super(instanceContainer, objectData);
+
+      this._animations = objectData.content.animations;
+      this._timeScale = objectData.content.timeScale;
+      this._opacity = objectData.content.opacity;
+      this._scale = objectData.content.scale;
+      this.jsonResourceName = objectData.content.jsonResourceName;
+      this.atlasResourceName = objectData.content.atlasResourceName;
+      this.imageResourceName = objectData.content.imageResourceName;
+      this._renderer = new gdjs.SpineRuntimeObjectRenderer(
+        this,
+        instanceContainer
+      );
+
+      // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+      this.onCreated();
+    }
+
+    getRendererObject(): pixi_spine.Spine | PIXI.Container {
+      return this._renderer.getRendererObject();
+    }
+
+    updateFromObjectData(
+      oldObjectData: SpineObjectData,
+      newObjectData: SpineObjectData
+    ): boolean {
+      super.updateFromObjectData(oldObjectData, newObjectData);
+
+      if (oldObjectData.content.opacity !== newObjectData.content.opacity) {
+        this.setOpacity(newObjectData.content.opacity);
+      }
+      if (oldObjectData.content.scale !== newObjectData.content.scale) {
+        this.setScale(newObjectData.content.scale);
+      }
+      if (oldObjectData.content.timeScale !== newObjectData.content.timeScale) {
+        this.setTimeScale(newObjectData.content.timeScale);
+      }
+
+      return true;
+    }
+
+    extraInitializationFromInitialInstance(
+      initialInstanceData: InstanceData
+    ): void {
+      const animationData = initialInstanceData.numberProperties?.find(
```

Comment:
We usually try to avoid the ?. operator, as something null/undefined is unexpected here. I think this should work:

```suggestion
      const animationData = initialInstanceData.numberProperties.find(
```